packages/api/src/routes/medical/internal-cq.ts

```diff
@@ -146,4 +149,20 @@ router.post(
   })
 );
 
+/**
+ * POST /internal/carequality/document-retrieval/response
+ *
+ * Receives a Document Retrieval response from the IHE Gateway
+ */
+router.post(
+  "/document-retrieval/response",
+  asyncHandler(async (req: Request, res: Response) => {
+    const docRetrieval = documentRetrievalResponseSchema.parse(req.body);
+
+    await handleIHEResponse({ type: IHEResultType.DOCUMENT_RETRIEVAL, response: docRetrieval });
```

Comment:
this is more of a comment on the way `handleIHEResponse` was defined in a previous PR, but I think defining the status based on whether or not operationOutcome.issue exists might not be accurate according to the schema?

according to the docs here: https://www.notion.so/metriport/XCA-ITI-38-Interface-Outgoing-DQ-0130d18144da45969bb80f179ea43335
 and here:
 https://www.notion.so/metriport/XCA-ITI-39-Interface-Outgoing-DR-593b06be795e46458eb1530dd9a7b822
 
 a successful response can still contain operationOutcomes with warnings or in the case of document Retrieval, it could have an error on one document ID but return the rest successfully. I think maybe it might make more sense to just check the relevant values for each request type to see if it exists to determine success or failure?
So `xcpdPatientId` for 55, `documenReference` for 38, and `documentReference` for 39. Maybe @ShamilN can chime in here though

Reply:
link to the code in question: https://github.com/metriport/metriport/blob/2712d90e1131c9315fa481464b6370f1467bf731/packages/api/src/command/medical/ihe-result/create-ihe-result.ts#L21

Reply:
Yea i took it as a flag to signify something went wrong but i think its a valid point to only fail if the whole request didnt work